libraries/lib-music-information-retrieval/MirUtils.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MirUtils.h"
+
+#include <cmath>
+#include <numeric>
+
+namespace MIR
+{
+namespace
+{
+constexpr auto bpmStdDev = 29.7953;
+constexpr auto pi = 3.141592653589793;
+
+constexpr bool IsPrimeDecompositionTwoThreeOnly(int n)
+{
+   while (n % 2 == 0)
+      n /= 2;
+   while (n % 3 == 0)
+      n /= 3;
+   return n == 1;
+}
+
+constexpr int CountThreesInPrimeDecomposition(int n)
+{
+   int count = 0;
+   while (n % 3 == 0)
+   {
+      ++count;
+      n /= 3;
+   }
+   return count;
+}
+
+// Function to generate numbers whose prime factorization contains only twos or
+// threes
+std::vector<int> GetPowersOf2And3(int lower, int upper)
+{
+   std::vector<int> result;
+   for (int i = lower; i <= upper; ++i)
+      if (IsPrimeDecompositionTwoThreeOnly(i))
+         result.push_back(i);
+   return result;
+}
+} // namespace
+
+std::vector<int> GetPossibleBarDivisors(int lower, int upper)
+{
+   auto result = GetPowersOf2And3(lower, upper);
+   result.erase(
+      std::remove_if(
+         result.begin(), result.end(),
+         [](int n) { return CountThreesInPrimeDecomposition(n) > 2; }),
```

Comment:
Or simply `n % 9 == 0`

Reply:
So really `GetPowersOf2And3` could generate just the powers of 2 and the triples of those

Reply:
`n % 27 == 0` - we hold musical rhythms with two triple levels for realistic (e.g. a swung 6/8). But thanks for the tip.